backend/modules/assistant/ito/difference/difference_agent.py

```diff
@@ -0,0 +1,133 @@
+import asyncio
+
+import pandas as pd
+from pydantic import BaseModel, Field
+
+from .query_engine import DecisionEnum, DiffQueryEngine
+from .question_generator import QuestionGenerator
+
+
+class ResponseType(BaseModel):
+    """Represents an ingredient and its associated decision."""
+
+    name: str | None = Field(description="Name of the ingredient")
+    detailed_answer: str | None = Field(description="Detailed answer with explanations")
+    decision: DecisionEnum | None = Field(
+        description="Decision made: authorized / to avoid / forbidden"
+    )
+
+
+class ContextType(BaseModel):
+    """Represents the context given and the specific category if needed."""
+
+    category: str = Field(description="Category of the context")
+    context: str = Field(description="Context to add to the query")
+
+
+"""
+
+"""
+
+
+class DifferenceAgent:
+    def __init__(
+        self,
+        diff_query_engine: DiffQueryEngine,
+        document_context: str | None = None,
+        question_generator: QuestionGenerator | None = None,
+    ):
+        self.document_context = document_context
+        self.question_generator = (
+            question_generator
+            if question_generator
+            else QuestionGenerator(document_context=document_context)
+        )
+        self.diff_query_engine = diff_query_engine
+
+        self.questions = None
+        self.generated_df = pd.DataFrame()
```

Comment:
Do we need to initiate an empty dataframe here just to override it or we can return an empty df ?

Reply:
So we can override it, and if there was an error we can access an empty df (if empty .. )